fedbiomed/researcher/federated_workflows/_experiment.py

```diff
@@ -835,7 +835,7 @@ def run_once(self, increase: bool = False, test_after: bool = False) -> int:
                           optim_aux_var=optim_aux_var
                           )
 
-        logger.info('Sampled nodes in round ' + str(self._round_current) + ' ' + str(job.nodes))
+        logger.info('Sampled nodes in round ' + str(self._round_current) + ' ' + str(training_nodes))
```

Comment:
Thanks for sharing this, after second thoughts I think the best option is:
- let's keep the getter, and use `job.nodes` instead of `training_nodes`
- let's remove the getter: `Job` is one shot and its members should not be modified from the outside after init

Reply:
@sharkovsky do you mean "let's keep the getter, let's remove the **setter**" ?

This option would be fine for me.

Reply:
yeah that's what I meant :(

Reply:
It should be fixed now. just let me know if it is better a property or a setter `job.nodes` or `job.nodes()`